### PR TITLE
HUD says when filters or gates are narrowing its counts

### LINKED ISSUE #1328: HUD in-view count should indicate active filters/gates, e.g. 'Showing 826 of 826 in view (1 filter applied)'
## Problem

The render-coverage HUD ("Showing 826 of 826 in view") reports counts **after** filters and analysis gates are applied, but gives no hint that any narrowing is active. When a gate or filter is restored from the saved configuration at load, the viewport can visibly contain far more objects than the HUD's denominator, and the display reads like data loss.

This just produced a real "is my dataset corrupted?" moment for an expert user: a saved 16-vertex lasso gate (Area × PECAM1, resolving to 72,925 of 708,983 annotations, 10.3%) made the HUD read "Showing 826 of 826 in view" in a viewport that visually holds thousands of objects. The existing "1" badge on the analysis palette icon was present the whole time and was not enough of a cue — the badge lives far from the count the user is actually reading.

## Proposal

Append an active-constraint suffix to the HUD label when any object-narrowing constraint is active, e.g.:

> Showing 826 of 826 in view **(1 filter applied)** — or **(3 filters applied)**

Details:
- Count all sources that narrow the object set: enabled analysis gates, tag filters, enabled property filters, ROI filters, and the annotation-ID filter — i.e. the union of what the app bar's Filters badge (`activeFilterCount`) and Analysis badge (`activeAnalysisGateCount`) count, so the three surfaces stay consistent.
- The suffix should be visually distinct (e.g. warning-tinted or underlined) and ideally clickable / hoverable: a tooltip listing the active constraints ("1 lasso gate on Area × PECAM1; 1 tag filter"), and a click opening the relevant panel (Filters or Analysis).
- Show the suffix in both stub (large-dataset) and client modes — the count is filtered in both.

## Pointers

- HUD component: `src/components/RenderCoverageIndicator.vue` (labels built in `src/utils/renderCoverage.ts`).
- Constraint counts: `activeFilterCount` and `activeAnalysisGateCount` in `src/App.vue` currently drive the palette-icon badges; extract the counting into a shared util so the HUD and badges can't drift.
- Related precedent: the analysis-badge comment in `App.vue` ("gates apply with the palette closed... a dataset could open already filtered with nothing on screen to say why") — this issue extends that reasoning to the HUD, which is where users actually look when they suspect missing data.

### PR DESCRIPTION
**Pull request by arjunrajlab**
Closes #1328.

## Problem

The render-coverage HUD prints counts computed **after** filters and analysis gates (`viewportAnnotationCount` / `viewportRenderedCount` come from the id set that already survived them), with nothing to say so. A saved lasso gate restored at load (Area × PECAM1, 72,925 of 708,983) made it read "Showing 826 of 826 in view" in a viewport that visibly held thousands — indistinguishable from data loss. The `1` badge on the Analysis palette icon was showing the whole time and did not prevent the report: it lives at the edge of the window, far from the number being read, and says nothing about the count it explains.

## What changed

**The HUD label carries an active-constraint suffix.**

> Showing 826 of 826 in view **(1 filter applied)**

- Warning-tinted, dotted-underlined, on the same line as the counts it explains.
- Tooltip (also its `aria-label`, since the visible text is only a count): *"Objects are narrowed by 1 lasso gate on Area × PECAM1; 1 tag filter. Click to open Analysis and Filters."*
- Clicking opens the panels owning the active constraints — Analysis first, then Filters, because Filters is a companion palette that hosts alongside Analysis and the other order would close what was just opened.
- Not gated on stub mode, since the counts are filtered in client mode too. It does **not** change *when* the HUD appears: the show rule is still stub mode or active downsampling. A small dataset with a filter and no downsampling never showed the misleading denominator, so forcing the HUD open there would be new noise rather than a fix.

**One list behind all three surfaces.** New `src/utils/activeConstraints.ts` decides what counts as narrowing the object set: `collectActiveConstraints` (pure data — each entry carries its `source`, `kind`, and enough detail to name it later), `countActiveConstraints`, and `describeConstraint` / `summarizeActiveConstraints` for phrasing. `activeFilterCount` and `activeAnalysisGateCount` keep their exact semantics (each badge still counts only what its own panel can show) and now just count that one list; the HUD reads both halves via `activeConstraintCount`. A new narrowing filter that skips the collector is now invisible on all three surfaces rather than on two — that trap is written into the doc and the skill.

Phrasing is deliberately separate from collecting: naming a property path needs the properties store, which shouldn't be pulled into a filters-store getter. It degrades rather than invents — a half-configured gate drops the plane instead of implying a one-dimensional gate, and an unresolvable property path drops the name.

**Reaching the palette registry.** The HUD is mounted deep inside `ImageViewer`, with no path to App.vue's palette state, so it asks through the store: `requestPaletteOpen([...])` sets `paletteOpenRequests`, App.vue opens each in order and clears the list. This generalizes the existing `isAnnotationPanelOpen` hatch used by the Timelapse panel. `TRequestablePalette` is a subset of App.vue's `PaletteId`, so a renamed palette is a compile error rather than a click that silently does nothing.

## Tests

`pnpm test` (3649 tests, 204 files), `pnpm lint:ci`, `pnpm tsc` and `pnpm build` all pass. 20 new tests:

- `src/utils/__tests__/activeConstraints.test.ts` — what counts (every filter kind; disabled rows and a region still being drawn excluded; a gate only when enabled **and** drawn **and** resolved), per-panel vs combined counts, and the phrasing edge cases.
- `src/components/RenderCoverageIndicator.test.ts` — suffix presence/absence, the full sentence including the space that template whitespace handling is free to eat, the tooltip mirrored into `aria-label`, both click targets, and the non-stub-mode case.
- `src/utils/__tests__/renderCoverage.test.ts` — the label itself, pluralized, and outside stub mode.
- `src/App.test.ts` — the request watcher opens every palette named then clears the request; an empty request is a no-op.

Two test-harness traps hit during this and are now recorded in `nimbus-frontend`: a `vi.mock` factory captures the spy it closes over (so `mockClear`, never reassign in `beforeEach`), and a plain-object store mock never fires a watcher (so App.vue's store mock is now `reactive` — otherwise both watcher tests would have passed vacuously).

**Not verified in a live browser** — no backend or dataset available in this environment. The visual claims rest on the component test and the CSS, so the tint and hover state are worth a glance when running locally.

## Docs

New `codebaseDocumentation/ACTIVE_CONSTRAINT_CUES.md` (problem, design, regression checklist), cross-referenced from `ANALYSIS_PANEL.md` and `SERVER_GATING.md` where the "each badge counts only what its own panel can show" prose lives.

🤖 Generated with [Claude Code](https://claude.com/claude-code)

---
_Generated by [Claude Code](https://claude.ai/code/session_0124dCot4gNFX7PYhXwt1Wn3)_

### CODE REVIEW

**Comment by arjunrajlab** [comment]
@codex review

File: src/utils/activeConstraints.ts [hunk 1]
```diff
@@ -0,0 +1,218 @@
...
+  if (input.onlyCurrentFrame) {
+    pushFilter("currentFrame");
+  }
+  if (input.selectionFilter.enabled) {
+    pushFilter("selection");
+  }
+  if (!input.showAnnotationsFromHiddenLayers) {
+    pushFilter("hiddenLayers");
+  }
+  for (const filter of input.propertyFilters.filter(isEnabled)) {
```
[the remark this answers is not in the document]
Comment (arjunrajlab) [reply src/utils/activeConstraints.ts]: Fixed in 85335bb9. `collectActiveConstraints` now applies the same no-op predicate as the filtering path: an enabled values-mode filter with an empty (or missing) values list is skipped, mirroring `filters.ts`'s pass-all treatment and the backend's `dropNoOpPropertyFilters`. Covered in the collector regression tests (`does not count an enabled values filter whose values list is empty` — empty, undefined, and repopulated states).

Swept the other collected kinds for the same shape: an empty tag filter genuinely narrows (non-exclusive hides everything; exclusive matches only untagged objects), empty selection/id-list filters narrow to zero, and the client's range filters always carry numeric bounds — so this was the only no-op. Note this also stops the Filters badge from counting the no-op row (all three surfaces count the one list); that semantic change is deliberate and documented in ACTIVE_CONSTRAINT_CUES.md.
